src/main/java/org/cef/browser/CefRenderer.java

```diff
@@ -119,79 +125,93 @@ protected void clearPopupRects() {
         original_popup_rect_.setBounds(0, 0, 0, 0);
     }
 
+    @SuppressWarnings("static-access")
     protected void onPaint(boolean popup, Rectangle[] dirtyRects, ByteBuffer buffer, int width, int height, boolean completeReRender) {
-        if(transparent_) // Enable alpha blending.
-            GlStateManager.enableBlend();
-
-        final int size = (width * height) << 2;
-        if(size > buffer.limit()) {
-            Log.warning("Bad data passed to CefRenderer.onPaint() triggered safe guards... (1)");
-            return;
+        if (transparent_) {
+            // Enable alpha blending.
+            glEnable(GL_BLEND);
         }
 
         // Enable 2D textures.
-        GlStateManager.enableTexture2D();
-        GlStateManager.bindTexture(texture_id_[0]);
+        glEnable(GL_TEXTURE_2D);
+
+        assert (texture_id_[0] != 0);
+        glBindTexture(GL_TEXTURE_2D, texture_id_[0]);
+
+        if (!popup) {
+            int old_width = view_width_;
```

Comment:
we should review our code for parity with JCEF in case they changed anything significant here